pkg/apis/v1/nodeclaim_conversion_test.go

```diff
@@ -0,0 +1,539 @@
+			}
+		})
+		It("should convert v1 nodeclaim startup taints", func() {
+			v1NodeClaim.Spec.StartupTaints = []v1.Taint{
+				{
+					Key:    "test-key-startup-1",
+					Value:  "test-value-startup-1",
+					Effect: v1.TaintEffectNoExecute,
+				},
+				{
+					Key:    "test-key-startup-2",
+					Value:  "test-value-startup-2",
+					Effect: v1.TaintEffectNoSchedule,
+				},
+			}
+			Expect(len(v1beta1NodeClaim.Spec.StartupTaints)).To(BeNumerically("==", 0))
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			for i := range v1NodeClaim.Spec.StartupTaints {
+				Expect(v1beta1NodeClaim.Spec.StartupTaints[i].Key).To(Equal(v1NodeClaim.Spec.StartupTaints[i].Key))
+				Expect(v1beta1NodeClaim.Spec.StartupTaints[i].Value).To(Equal(v1NodeClaim.Spec.StartupTaints[i].Value))
+				Expect(v1beta1NodeClaim.Spec.StartupTaints[i].Effect).To(Equal(v1NodeClaim.Spec.StartupTaints[i].Effect))
+			}
+		})
+		It("should convert v1 nodeclaim requirements", func() {
+			v1NodeClaim.Spec.Requirements = []NodeSelectorRequirementWithMinValues{
+				{
+					NodeSelectorRequirement: v1.NodeSelectorRequirement{
+						Key:      v1.LabelArchStable,
+						Operator: v1.NodeSelectorOpExists,
+					},
+					MinValues: lo.ToPtr(0),
+				},
+				{
+					NodeSelectorRequirement: v1.NodeSelectorRequirement{
+						Key:      CapacityTypeLabelKey,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{CapacityTypeSpot},
+					},
+					MinValues: lo.ToPtr(0),
+				},
+			}
+			Expect(len(v1beta1NodeClaim.Spec.Requirements)).To(BeNumerically("==", 0))
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			for i := range v1NodeClaim.Spec.Requirements {
+				Expect(v1beta1NodeClaim.Spec.Requirements[i].Key).To(Equal(v1NodeClaim.Spec.Requirements[i].Key))
+				Expect(v1beta1NodeClaim.Spec.Requirements[i].Operator).To(Equal(v1NodeClaim.Spec.Requirements[i].Operator))
+				Expect(v1beta1NodeClaim.Spec.Requirements[i].Values).To(Equal(v1NodeClaim.Spec.Requirements[i].Values))
+			}
+		})
+		It("should convert v1 nodeclaim resources", func() {
+			v1NodeClaim.Spec.Resources = ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("1"),
+					v1.ResourceMemory: resource.MustParse("134G"),
+				},
+			}
+			Expect(v1beta1NodeClaim.Spec.Resources.Requests).To(BeNil())
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			for key := range v1NodeClaim.Spec.Resources.Requests {
+				Expect(v1NodeClaim.Spec.Resources.Requests[key]).To(Equal(v1beta1NodeClaim.Spec.Resources.Requests[key]))
+			}
+		})
+		Context("NodeClassRef", func() {
+			It("should convert v1 nodeclaim template nodeClassRef", func() {
+				v1NodeClaim.Spec.NodeClassRef = &NodeClassReference{
+					Kind:  "fake-cloudprovider-kind",
+					Name:  "nodeclass-test",
+					Group: "fake-cloudprovider-group",
+				}
+				Expect(v1beta1NodeClaim.Spec.NodeClassRef).To(BeNil())
```

Comment:
Why is this here? Can we just assume that it's not set to start?

Reply:
We can assume this. I was trying explicitly call that out